README.md
# Championship Project

*About:* Sa se realizeze o aplicatie client/server ce va administra diferite campionate. Aplicatia server va avea urmatoarele functionalitati: inregistrarea utilizatorilor(diferite tipuri de utilizatori: obisnuiti, administratori). Toate comenzile vor fi restrictionate de sectiunea de logare. Dupa logare utilizatorii vor putea primi informatii despre ultimele campionate inregistrate, avand posibilitatea de a se inscrie la ele. Comenzile ce trebuie implementate: inregistrarea unui campionat, specificarea jocului, numarului de jucatori, diferite reguli sau structuri de campionat(single-elimination, double elimination), modul de extragere a partidelor(deciderea partidelor), inregistrarea unui utiliziator intr-un campionat, utilizatorul va fi informat via e-mail daca a fost acceptat in campionatul respectiv, si va primi informatii aditionale despre partidele sale (ora, adversarul, etc...) Utilizatorul are posibilitatea sa reprogrameze o sesiune de joc. Utilizatorii administratori vor detine un istoric al scorurilor partidelor.

Comenzi:
 gcc baze.c -o baze -lsqlite3
 ./baze
 make
 ./client

 For emailing you'll need to install on your local device:
 sudo apt-get install libcurl4-nss-dev

history.txt


CHAMPIONSHIPS HISTORY                   
----------------------------------------
Name: International Chess
History: 12/4/15, 17/7/20
Last winner: Andrei
----------------------------------------
Name: National Chess
History: 12/4/17, 7/3/20
Last winner: Monica
----------------------------------------
Name: Tennis for teams
History: 12/12/19
Last winner: Team Australia
----------------------------------------
Name: Tennis
History: none
Last winner: none
----------------------------------------
Name: Football
History: 12/12/13
Last winner: Team Denmark
----------------------------------------


client.c
#include <sys/types.h>
#include <arpa/inet.h>
#include <sys/socket.h>
#include <netinet/in.h>
#include <errno.h>
#include <unistd.h>
#include <stdio.h>
#include <stdlib.h>
#include <netdb.h>
#include <string.h>
#include <sqlite3.h>

int port = 2009;
sqlite3 *db;
char *zErrMsg = 0;

void read_file(char command[])
{
  FILE *fp;
  if(strstr(command,"History")!=NULL){
    fp = fopen("history.txt","r");
  }
  else{
    fp = fopen("championships.txt","r");
  }
  char buffer[256];
  while (fgets(buffer, 256, fp)){
        printf("%s", buffer);
  }
  printf("\n\n\n");
  fclose(fp);
}

int main ()
{
  int sd;			                // descriptorul de socket
  struct sockaddr_in server;	// structura folosita pentru conectare 
  char command[200];

  /* cream socketul */
  if ((sd = socket (AF_INET, SOCK_STREAM, 0)) == -1)
    {
      printf("Eroare la socket().\n");
      exit(1);
    }

  /* familia socket-ului */
  server.sin_family = AF_INET;
  /* adresa IP a serverului */
  server.sin_addr.s_addr = inet_addr("127.0.0.1");
  /* portul de conectare */
  server.sin_port = htons (port);
  
  if (connect (sd, (struct sockaddr *) &server,sizeof (struct sockaddr)) == -1)
    {
      printf ("[client]Eroare la connect().\n");
      exit(1);
    }

  while(1){
  
  printf ("CLIENT:\t");
  fflush (stdout);
  read (0, command, sizeof(command));

  char nume_edit[200];
  //printf("[client] Am citit %s\n",command);
  if (write (sd,&command,sizeof(command)) <= 0)
    {
      printf("[client]Eroare la write() spre server.\n");
      exit(1);
    }

  if (read (sd, &command,sizeof(command)) < 0)
    {
      printf ("[client]Eroare la read() de la server.\n");
      exit(1);
    }
    
  //verificam ce a trimis serverul
  printf ("SERVER:\t%s\n", command);
  if(strstr(command,"The list of championships")!=NULL || strstr(command,"History")!=NULL){
      read_file(command);
  }
  else if(strstr(command,"Goodbye")!=NULL){
    close (sd);
    break;
  }
  memset(command, 0, sizeof(command));
  }
  sqlite3_close(db);
}

baze.c
#include <stdio.h>
#include <stdlib.h>
#include <sqlite3.h> 

static int callback(void *NotUsed, int argc, char **argv, char **azColName) {
   int i;
   for(i = 0; i<argc; i++) {
      printf("%s = %s\n", azColName[i], argv[i] ? argv[i] : "NULL");
   }
   return 0;
}

int main(int argc, char* argv[]) {
   sqlite3 *db;
   char *zErrMsg = 0;
   int rc;
   char *sql;

   /* Open database */
   rc = sqlite3_open("test.db", &db);
   
   if( rc ) {
      fprintf(stderr, "Can't open database: %s\n", sqlite3_errmsg(db));
      return(0);
   } else {
      fprintf(stdout, "Opened database successfully\n");
   }

   /* Create SQL statement */

   sql = "CREATE TABLE ACCOUNTS(" \
      "USERNAME CHAR(10) PRIMARY KEY NOT NULL,"\
      "PASSWORD TEXT NOT NULL,"\
      "NAME TEXT NOT NULL,"\
      "ADMIN INT NOT NULL,"\
      "STRONG TEXT,"\
      "WEAK TEXT,"\
      "EMAIL TEXT NOT NULL);";

   /* Execute SQL statement */
   rc = sqlite3_exec(db, sql, callback, 0, &zErrMsg);
   
   if( rc != SQLITE_OK ){
      printf("SQL error: %s\n", zErrMsg);
   } else {
      printf("Table created successfully\n");
   }

   sql = "CREATE TABLE CHAMPIONSHIPS(" \
      "NAME TEXT PRIMARY KEY NOT NULL,"\
      "TYPE TEXT NOT NULL,"\
      "NB_PLAYERS INT NOT NULL,"\
      "NR_PARTICIPANTI INT,"\
      "PARTICIPANTI TEXT,"\
      "STRUCTURE TEXT NOT NULL,"\
      "HISTORY TEXT,"\
      "WINNER TEXT,"\
      "GAMES DATE,"\
      "ORA TEXT,"\
      "LOCATION TEXT,"\
      "DESC TEXT);";

   /* Execute SQL statement */
   rc = sqlite3_exec(db, sql, callback, 0, &zErrMsg);
   
   if( rc != SQLITE_OK ){
      printf("SQL error: %s\n", zErrMsg);
   } else {
      printf("Table created successfully\n");
   }

   sql = "INSERT INTO ACCOUNTS (USERNAME,PASSWORD, NAME,ADMIN,STRONG, WEAK, EMAIL) "  \
         "VALUES ('daria','daria', 'Daria', 1,'tennis','football',  'dariae9@yahoo.com'); " \
         "INSERT INTO ACCOUNTS (USERNAME,PASSWORD, NAME,ADMIN,STRONG, WEAK,  EMAIL) "  \
         "VALUES ('paul','paul', 'Paul', 0,'tennis','chess', 'dariae9@yahoo.com'); " \
         "INSERT INTO ACCOUNTS (USERNAME,PASSWORD, NAME, ADMIN, STRONG, WEAK,  EMAIL) "  \
         "VALUES ('andreea','andreea', 'Andreea', 1,'football','tennis',  'dariae9@yahoo.com'); " \
         "INSERT INTO ACCOUNTS (USERNAME,PASSWORD, NAME, ADMIN, STRONG, WEAK,  EMAIL) "  \
         "VALUES ('stefan','stefan', 'Stefan', 0,'chess', 'football',  'dariae9@yahoo.com'); " \
         "INSERT INTO ACCOUNTS (USERNAME, PASSWORD, NAME,ADMIN,STRONG, WEAK,  EMAIL) "  \
         "VALUES ('mihai', 'mihai', 'Mihai', 1,'tennis','chess' , 'dariae9@yahoo.com'); " \
         "INSERT INTO CHAMPIONSHIPS (NAME, TYPE, NB_PLAYERS, NR_PARTICIPANTI, PARTICIPANTI, STRUCTURE, HISTORY, WINNER, GAMES, ORA, LOCATION, DESC) "  \
         "VALUES ('International Chess', 'chess', 2, 1,'mihai', 'one round','12/4/15, 17/7/20','Andrei', '2/2/27','12:00','Romania, Bucharest','Chess is an abstract strategy game and involves no hidden information. This championship requires experience'); " \
         "INSERT INTO CHAMPIONSHIPS (NAME, TYPE, NB_PLAYERS, NR_PARTICIPANTI, PARTICIPANTI, STRUCTURE,HISTORY, WINNER, GAMES, ORA,LOCATION, DESC) "  \
         "VALUES ('National Chess', 'chess', 4,1,'andrei', 'three rounds', '12/4/17, 7/3/20','Monica', '4/4/23','10:00','Romania, Iasi','Chess is an abstract strategy game and involves no hidden information. This championship requires experience'); " \
         "INSERT INTO CHAMPIONSHIPS (NAME, TYPE, NB_PLAYERS, NR_PARTICIPANTI, PARTICIPANTI, STRUCTURE, HISTORY, WINNER, GAMES, ORA, LOCATION, DESC) "  \
         "VALUES ('Tennis for teams', 'tennis', 4, 0,'', '1 round','12/12/19','Team Australia', '23/5/24', '13:00','Romania, Cluj','Tennis is a racket sport that can be played individually against a single opponent: singles '); " \
         "INSERT INTO CHAMPIONSHIPS (NAME, TYPE, NB_PLAYERS, NR_PARTICIPANTI, PARTICIPANTI, STRUCTURE, HISTORY, WINNER, GAMES, ORA, LOCATION, DESC) "  \
         "VALUES ('Tennis', 'tennis', 2, 1,'andreea', '1 round','none','none', '15/3/23','16:00','UK, London','Tennis is a racket sport that can be played between two teams of two players each : doubles '); " \
         "INSERT INTO CHAMPIONSHIPS (NAME, TYPE, NB_PLAYERS, NR_PARTICIPANTI, PARTICIPANTI, STRUCTURE, HISTORY, WINNER, GAMES, ORA, LOCATION, DESC) "  \
         "VALUES ('Football', 'football', 2,1,'ana', '5 rounds', '12/12/13' , 'Team Denmark', '12/12/25','20:00','Germany, Berlin','Football is a game involving two teams of 11 players who try to maneuver the ball into goal without using their hands or arms ' ); " ;
         

   /* Execute SQL statement */
   rc = sqlite3_exec(db, sql, callback, 0, &zErrMsg);
   
   if( rc != SQLITE_OK ){
      printf("SQL error: %s\n", zErrMsg);
   } else {
      printf("Records created successfully\n");
   }
   sqlite3_close(db);
   return 0;
}